File: LeetCodeRecord/MethodFirst/349-Intersection.java
package MethodFirst;

import java.lang.reflect.Array;
import java.security.DrbgParameters.Reseed;
import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

class Intersection {
    public int[] intersection(int[] nums1, int[] nums2) {
        Set<Integer> nums1Set = new HashSet<Integer>();
        for (int num : nums1) {
            nums1Set.add(num);
        }

        Set<Integer> resSet = new HashSet<Integer>();
        for (int num : nums2) {
            if (nums1Set.contains(num)) {
                resSet.add(num);
            }
        }
        
        int[] res = new int[resSet.size()];
        int i = 0;
        for (int num : resSet) {
            res[i++] = num;
        }

        return res;
    }    
}

File: LeetCodeRecord/ACFirst-Golang/349-Intersection.go
package main

func intersection(nums1 []int, nums2 []int) []int {
	set1 := make(map[int]bool)
	for _, i := range nums1 {
		set1[i] = true
	}

	set2 := make(map[int]bool)
	for _, i := range nums2 {
		if _, ok := set1[i]; ok {
			set2[i] = true
		}
	}

	res := []int{}
	for i, _ := range set2 {
		res = append(res, i)
	}

	return res
}
